### PR TITLE
New post: stable diffusion inference on Intel CPUs

### PR DESCRIPTION
**Pull request by juliensimon**

### CODE REVIEW

**Comment by juliensimon** [comment]
Shared with Intel, waiting for feedback.

File: stable_diffusion_inference_intel.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+Out of the box, we can see that Sapphire Rapids CPUs are quite faster without any code change!
+
+Now, let's accelerate!
+
+## Optimum Intel and OpenVINO
+
+[Optimum Intel](https://huggingface.co/docs/optimum/intel/index) accelerates end-to-end pipelines on Intel architectures. Its API is extremely similar to the vanilla [Transformers](https://huggingface.co/docs/transformers/index) API, making it trivial to adapt existing code.
+
+Optimum Intel supports [OpenVINO](https://docs.openvino.ai/latest/index.html), an Intel open-source toolkit for high-performance inference. Let's update the code above: we only need to replace `StableDiffusionPipeline` with `OVStableDiffusionPipeline`:
+
```
Comment (echarlaix) [inline stable_diffusion_inference_intel.md:76]: Optimum can be installed using pip as follows:

```
pip install optimum[openvino]
```

File: stable_diffusion_inference_intel.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+The average latency is **32.6 seconds**. As demonstrated by this [Intel Space](https://huggingface.co/spaces/Intel/Stable-Diffusion-Side-by-Side), the same code runs on a previous generation Intel Xeon (code name Ice Lake) in about 45 seconds. 
+
+Out of the box, we can see that Sapphire Rapids CPUs are quite faster without any code change!
+
+Now, let's accelerate!
+
+## Optimum Intel and OpenVINO
+
+[Optimum Intel](https://huggingface.co/docs/optimum/intel/index) accelerates end-to-end pipelines on Intel architectures. Its API is extremely similar to the vanilla [Transformers](https://huggingface.co/docs/transformers/index) API, making it trivial to adapt existing code.
```
Comment (echarlaix) [inline stable_diffusion_inference_intel.md:73]:
```suggestion
[Optimum Intel](https://huggingface.co/docs/optimum/intel/index) accelerates end-to-end pipelines on Intel architectures. Its API is extremely similar to the [Diffusers](https://huggingface.co/docs/diffusers/index) API, making it trivial to adapt existing code.
```

**Review by echarlaix — COMMENTED** [review]
It looks great @juliensimon, just left a couple of comments

File: stable_diffusion_inference_intel.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
+---
+title: "Accelerating Stable Diffusion Inference on Intel CPUs"
+thumbnail: /blog/assets/xxx/01.png
+authors:
+- user: juliensimon
+- user: ellacharlaix
+---
+
+<h1>
+Accelerating Stable Diffusion Inference on Intel CPUs</h1>
```
Comment (julien-c) [inline stable_diffusion_inference_intel.md:10]:
```suggestion
# Accelerating Stable Diffusion Inference on Intel CPUs
```

File: stable_diffusion_inference_intel.md
Comment (julien-c) [inline stable_diffusion_inference_intel.md:1]: i would use dashes not underscores

Comment (julien-c) [reply stable_diffusion_inference_intel.md:1]: also maybe remove `inference`

File: stable_diffusion_inference_intel.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+```
+virtualenv sd_inference
+source sd_inference/bin/activate
+pip install transformers diffusers accelerate torch==1.13.1
+```
+
+Then, we write a simple benchmarking function that repeatedly runs inference, and returns the average latency for a single-image generation.
+
+```
```
Comment (julien-c) [inline stable_diffusion_inference_intel.md:38]:
```suggestion
```python
```

**Review by julien-c — COMMENTED** [review]
cool post!

File: stable_diffusion_inference_intel.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+## Optimum Intel and OpenVINO
+
+[Optimum Intel](https://huggingface.co/docs/optimum/intel/index) accelerates end-to-end pipelines on Intel architectures. Its API is extremely similar to the vanilla [Transformers](https://huggingface.co/docs/transformers/index) API, making it trivial to adapt existing code.
+
+Optimum Intel supports [OpenVINO](https://docs.openvino.ai/latest/index.html), an Intel open-source toolkit for high-performance inference. Let's update the code above: we only need to replace `StableDiffusionPipeline` with `OVStableDiffusionPipeline`:
+
+```
+from optimum.intel.openvino import OVStableDiffusionPipeline
+...
+ov_pipe = OVStableDiffusionPipeline.from_pretrained(model_id, export=True).to("cpu")
```
[the remark this answers is not in the document]
Comment (juliensimon) [reply stable_diffusion_inference_intel.md]: done

File: stable_diffusion_inference_intel.md [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+latency = elapsed_time(ov_pipe, prompt)
+print(latency)
+```
```
[the remark this answers is not in the document]
Comment (juliensimon) [reply stable_diffusion_inference_intel.md:84]: done

File: stable_diffusion_inference_intel.md (hunk 1 above)
Comment (juliensimon) [reply stable_diffusion_inference_intel.md:76]: done

File: stable_diffusion_inference_intel.md (hunk 2 above)
Comment (juliensimon) [reply stable_diffusion_inference_intel.md:73]: makes sense, done

File: stable_diffusion_inference_intel.md
Comment (juliensimon) [reply stable_diffusion_inference_intel.md:1]: there's a fine-tuning post coming :)

**Review by echarlaix — APPROVED** [review]

File: stable-diffusion-inference-intel.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+```python
+from optimum.intel.openvino import OVStableDiffusionPipeline
+...
+ov_pipe = OVStableDiffusionPipeline.from_pretrained(model_id, export=True)
+
+latency = elapsed_time(ov_pipe, prompt)
+print(latency)
+```
+
+OpenVINO automatically optimizes the model for the `bfloat16` format. Thanks to this, the average latency is now **16.7 seconds**, a sweet 2x speedup.q
```
[the remark this answers is not in the document]
Comment (juliensimon) [reply stable-diffusion-inference-intel.md:93]: ok

**Pull request merged**